ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/Situation.java

```diff
@@ -0,0 +1,299 @@
+     * CRAC result variant will be created but empty as well as sensitivity computation results variant.
+     *
+     * @return The situation variant ID of the newly created variant.
+     */
+    String createVariant() {
+        return createVariantFromNetworkVariant(network.getVariantManager().getWorkingVariantId());
+    }
+
+    /**
+     * This methods creates a variant from an already existing situation variant. It is advised to use this method
+     * to create a new variant. Network variant will be copied from the reference variant, CRAC result variant will be
+     * created but empty and sensitivity computation results will be copied from the reference variant.
+     *
+     * @param referenceVariantId: Already existing situation variant ID.
+     * @return The situation variant ID of the newly created variant.
+     * @throws FaraoException if referenceVariantId is not an existing variant of the situation.
+     */
+    String cloneVariant(String referenceVariantId) {
+        if (!variantIds.contains(referenceVariantId)) {
+            throw new FaraoException(String.format("Situation does not contain %s  as reference variant", referenceVariantId));
+        }
+        String situationVariantId = createVariantFromNetworkVariant(referenceVariantId);
+        systematicSensitivityAnalysisResultMap.put(situationVariantId, getSystematicSensitivityAnalysisResult());
+        return situationVariantId;
+    }
+
+    /**
+     * This method deletes a variant according to its ID. If the working variant is the variant to be deleted nothing
+     * would be done.
+     *
+     * @param variantId: Variant ID that is required to delete.
+     * @param keepCracResult: If true it will delete the variant as situation variant and the related network variant
+     *                      but it will keep the crac variant.
+     * @throws FaraoException if variantId is not an existing variant of the situation.
+     */
+    void deleteVariant(String variantId, boolean keepCracResult) {
+        if (!variantIds.contains(variantId)) {
+            throw new FaraoException(String.format(UNKNOWN_VARIANT, variantId));
+        }
+        if (!variantId.equals(workingVariantId)) {
+            network.getVariantManager().removeVariant(variantId);
+            systematicSensitivityAnalysisResultMap.remove(variantId);
+            if (!keepCracResult) {
+                crac.getExtension(ResultVariantManager.class).deleteVariant(variantId);
+            }
+            variantIds.remove(variantId);
+        }
+    }
+
+    /**
+     * This method clear all the situation variants with their related variants in the different data objects. It
+     * enables to keep some CRAC result variants as it is results.
+     *
+     * @param remainingVariantsForCrac IDs of the variants we want to keep the results in CRAC
+     */
+    void clear(List<String> remainingVariantsForCrac) {
```

Comment:
maybe rename the method so it's more explicit that the listed variants are the ones which are not cleared